### PR TITLE
Update troubleshooting guide, fix broken links

### PR DESCRIPTION
**Pull request by paulromano**
This PR updates our troubleshooting guide to include better information on how users can debug geomtery errors (namely, by using the openmc-plotter application). I've also fixed a whole bunch of broken links in our documentation and removed a small section talking about converting data to MATLAB.

### CODE REVIEW

File: docs/source/methods/neutron_physics.rst [hunk 1]
```diff
@@ -359,7 +359,7 @@ secondary energy and angle sampling.
 For a reaction with secondary products, it is necessary to determine the
 outgoing angle and energy of the products. For any reaction other than elastic
 and level inelastic scattering, the outgoing energy must be determined based on
-tabulated or parameterized data. The `ENDF-6 Format <endf102>`_ specifies a
+tabulated or parameterized data. The `ENDF-6 Format` specifies a
```
Comment (pshriwise) [inline docs/source/methods/neutron_physics.rst:362]: Was this supposed to be a link like it is in the change below?
```suggestion
tabulated or parameterized data. The `ENDF-6 Format`_ specifies a
```

File: docs/source/usersguide/processing.rst [hunk 2]
```diff
@@ -34,30 +34,20 @@ as requested; it is used in many of the provided plotting utilities, OpenMC's
...
-explore and plot structured mesh tallies for any scores and filter bins.
+The `example notebook`_ also demonstrates how to plot a structured mesh tally in
+two dimensions using the Python API. One can also use the :ref:`scripts_plot`
+script which provides an interactive GUI to explore and plot structured mesh
+tallies for any scores and filter bins.
 
 .. image:: ../_images/plotmeshtally.png
    :width: 400px
 
-Getting Data into MATLAB
```
Comment (pshriwise) [inline docs/source/usersguide/processing.rst:51]: Was this section no longer relevant/useful?

**Review by pshriwise — CHANGES_REQUESTED** [review]
Thanks for these updates @paulromano! A couple of small questions, but then it's ready to go.

File: docs/source/methods/neutron_physics.rst (hunk 1 above)
Comment (paulromano) [reply docs/source/methods/neutron_physics.rst:362]: Yes, it absolutely was. Thanks for catching this!

File: docs/source/usersguide/processing.rst (hunk 2 above)
Comment (paulromano) [reply docs/source/usersguide/processing.rst:51]: I personally don't recommend people to use MATLAB given the capabilities available in Python and its open source ecosystem. If someone really thinks we ought to leave this in though, now is the time to speak up.

**Review by pshriwise — APPROVED** [review]
Thanks @paulromano!

**Pull request merged**